tests/utils/test_pidlock.py

```diff
@@ -0,0 +1,164 @@
+"""Tests for the PidFileLock utility."""
+
+import errno
+import logging
+import os
+import shutil
+from pathlib import Path
+from unittest.mock import Mock, patch
+
+import pytest
+
+# Need to mock atexit.unregister, which is called inside PidFileLock.release(). This is crucial for isolated testing.
+import atexit 
+
+from docbuild.utils.pidlock import PidFileLock
+
+log = logging.getLogger(__name__)
+
+# --- Helper Fixtures and Mocks ---
+
+@pytest.fixture
+def mock_pid_running(monkeypatch):
+    """
+    Mocks os.kill to simulate a running process (PID exists).
+    os.kill(pid, 0) succeeds if the process exists.
+    """
+    def mock_kill(pid, sig):
+        if pid == 12345:  # Mock PID of a running process used in the test
+            return
+        # For any other PID, assume dead (simulating ESRCH)
+        raise OSError(errno.ESRCH, "No such process")
+    
+    monkeypatch.setattr(os, "kill", mock_kill)
+
+@pytest.fixture
+def mock_pid_dead(monkeypatch):
+    """
+    Mocks os.kill to simulate a dead process (PID does not exist).
+    """
+    def mock_kill(pid, sig):
+        # Always raise ESRCH (No such process) for any checked PID
+        raise OSError(errno.ESRCH, "No such process")
+        
+    monkeypatch.setattr(os, "kill", mock_kill)
+
+
+@pytest.fixture
+def lock_setup(tmp_path: Path) -> tuple[Path, Path]:
+    """Provides a resource path and the temporary lock directory."""
+    resource_path = tmp_path / "env.production.toml"
+    resource_path.touch()  # The resource path must exist
+    lock_dir = tmp_path / "locks"
+    return resource_path, lock_dir
+
+# --- Test Cases ---
+
+def test_acquire_and_release_success(lock_setup):
+    """Test successful lock acquisition and release using a context manager."""
+    resource_path, lock_dir = lock_setup
+    
+    current_pid = os.getpid()
+
+    lock = PidFileLock(resource_path, lock_dir)
+    assert not lock.lock_path.exists()
+    
+    # Acquire lock via context manager
+    with lock:
+        assert lock._lock_acquired is True
```

Comment:
After adding the property, we should use it. 😉 


```suggestion
        assert lock.lock is True
```